main/php_odbc_utils.c

```diff
@@ -0,0 +1,120 @@
+/*
+   +----------------------------------------------------------------------+
+   | Copyright (c) The PHP Group                                          |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | https://www.php.net/license/3_01.txt                                 |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Calvin Buckley <calvin@cmpct.info>                          |
+   +----------------------------------------------------------------------+
+*/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "php.h"
+
+/*
+ * This files contains functions shared between ext/pdo_odbc and ext/odbc,
+ * relating to i.e. connection string quoting rules.
+ *
+ * The declarations are PHPAPI due to being available for shared/static
+ * versions.
+ */
+
+/**
+ * Determines if a string matches the ODBC quoting rules.
+ *
+ * A valid quoted string begins with a '{', ends with a '}', and has no '}'
+ * inside of the string that aren't repeated (as to be escaped).
+ *
+ * These rules are what .NET also follows.
+ */
+PHPAPI bool php_odbc_connstr_is_quoted(const char *str)
+{
+	if (!str) {
```

Comment:
I'm not sure whether we should check this; we may well push the obligation to provide an actual string to the caller (i.e. require this as precondition).